IndomitableIbises/Bangazon-Sprint-3
# Added Category Model/View/Temp Template

## Link to Ticket
References #26 

## Description of Proposed Changes
- Added **Category** model
- Added **list_categories** view
- Added **category.html** template
- Added **Category URL**
- Fixed **list_products**

## Steps to Test

Outline the steps to test

```sh
git fetch --all
git checkout RC-Category
```

1. Delete **migrations**, **db.sqlite3**
1. Ensure server is **not** running and db browser is **not** open
1. `python manage.py makemigrations`
1. `python manage.py migrate`
1. `python manage.py runserver
1. Open **db browser* and check to see if **category** table exists and add a **couple** of categories
1. You **do not** have to register or login to view categories
1. Go to this URL: `http://127.0.0.1:8000/categories`
1.  You should see the categories you created.



## Impacted Areas in Application

List general components of the application that this PR will affect:

* list_products.py
* list_categories.py
* category.py
* category.html
* urls.py

## Mentions @username

@paulzimmclay @sirwin4 @hayleylandsberg @eagobert 

## Definition of Done

1. The project must be fully documented. This includes the following:
   1. Complete README that documents the steps to install the code, how to install any dependencies, or system configuration needed.
   1. Every class must be documented with purpose, author, and methods.
   1. Every method must be documented with purpose and argument list - which itself must contain a short purpose for each argument.
1. The project must be able to run fully, and without errors, on each teammate's system.
1. Fulfills every requirement.
Every line of code has been peer reviewed.
1. For projects that require unit testing, core functionality must be identified and have at least one test for each
